File: king-admin-java/src/main/java/com/oukingtim/mapper/SysMenuMapper.java
package com.oukingtim.mapper;

import com.baomidou.mybatisplus.mapper.BaseMapper;
import com.oukingtim.domain.SysMenu;

import java.util.List;

/**
 * Created by oukingtim
 */
public interface SysMenuMapper extends BaseMapper<SysMenu> {

    /**
     * 根据用户id获取菜单
     * @param userId
     * @return
     */
    List<SysMenu> selectMenuByUserId(String userId);

    /**
     * 根据用户id查询权限
     * @param userId
     * @return
     */
    List<String> getPermissions(String userId);
}

File: king-admin-kotlin/src/main/kotlin/com/oukingtim/mapper/SysMenuMapper.kt
package com.oukingtim.mapper

import com.baomidou.mybatisplus.mapper.BaseMapper
import com.oukingtim.domain.SysMenu

/**
 * Created by oukingtim
 */
interface SysMenuMapper :BaseMapper<SysMenu> {

    /**
     * 根据用户id获取菜单
     * @param userId
     * *
     * @return
     */
    fun selectMenuByUserId(userId: String): List<SysMenu>

    /**
     * 根据用户id查询权限
     * @param userId
     * *
     * @return
     */
    fun getPermissions(userId: String): List<String>
}